config/logger/logging_test.go

```diff
@@ -65,3 +70,66 @@ func TestDefaultFormatterPanic(t *testing.T) {
 	}()
 	logrus.Panicln("Panic Test")
 }
+
+func TestFileLogger(t *testing.T) {
+	// initialize logger
+	filename := "/tmp/test_log"
+	testLog := &Config{
+		Level:  "DEBUG",
+		Format: "text",
+		Output: filename,
+	}
+	err := testLog.Init()
+	if err != nil {
+		t.Errorf("Did not expect error: %v", err)
+	}
+
+	testContainsLog := func(fname string, log string, t *testing.T) {
+		content, err := ioutil.ReadFile(fname)
+		if err != nil {
+			t.Errorf("Did not expect error: %v", err)
+		}
+		if len(content) == 0 {
+			t.Error("could not write log to file")
+		}
+		logs := string(content)
+		if !strings.Contains(logs, log) {
+			t.Errorf("expected log file to contain '%s', got '%s'", log, logs)
+		}
+	}
+
+	// write a log message
+	logMsg := "this is a test"
+	logrus.Info(logMsg)
+	testContainsLog(filename, logMsg, t)
+
+	// rotate the log
+	rotatedFilename := fmt.Sprintf("%s.1", filename)
+	err = os.Rename(filename, rotatedFilename)
+	if err != nil {
+		t.Errorf("Did not expect error: %v", err)
+	}
+	logMsg = "this is another test"
+	logrus.Info(logMsg)
+	testContainsLog(rotatedFilename, logMsg, t)
+
+	// reopen file
+	syscall.Kill(os.Getpid(), syscall.SIGUSR1)
+	time.Sleep(1*time.Second)
```

Comment:
Everything about this unit test feels right except that its adding 3 or 4 seconds to a full 15 second unit suite run. We should keep this unit test but move just the "reopen file" section into an integration test. 

I've begun looking into this but also wouldn't mind a PR enhancement from the author's end. A good integration test to look at would be `integration_tests/tests/test_logging/` and see what can be mimicked for the log re-open behavior. 

Fair enough?